polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -220,110 +220,117 @@ Value getNdItemArgument(FunctionOpInterface func) {
-                            const sycl::ReqdWorkGroupSize &reqdWorkGroupSize) {
+/// Get the shared memory needed by \p accTy in bytes.
+unsigned getReqdSharedMemory(sycl::AccessorType accTy,
+                             const sycl::ReqdWorkGroupSize &reqdWorkGroupSize) {
   assert(!reqdWorkGroupSize.empty() && "Expecting non-empty reqdWorkGroupSize");
+
   unsigned elemSize = accTy.getType().getIntOrFloatBitWidth() / 8;
-  unsigned memrefReqdLocalMemory = elemSize;
+  unsigned memrefReqdSharedMemory = elemSize;
   const unsigned numDims = accTy.getDimension();
   for (unsigned dim = 0; dim < numDims; ++dim) {
-    memrefReqdLocalMemory *= reqdWorkGroupSize[dim];
+    memrefReqdSharedMemory *= reqdWorkGroupSize[dim];
   }
-  return memrefReqdLocalMemory;
+  return memrefReqdSharedMemory;
 }
 
-/// Get the require local memory for memrefs in \p loopToSharedMemref, i.e, for
-/// each kernel. If there are multiple loops in the kernel that require local
+/// Get the require shared memory for memrefs in \p loopToSharedMemref, i.e, for
+/// each kernel. If there are multiple loops in the kernel that require shared
 /// memory, it returns the maximum amount required by any of them.
 Optional<unsigned>
-getReqdLocalMemory(const DenseMap<LoopLikeOpInterface, SetVector<Operation *>>
-                       &loopToSharedMemref,
-                   const sycl::ReqdWorkGroupSize &reqdWorkGroupSize) {
+getReqdSharedMemory(const DenseMap<LoopLikeOpInterface, SetVector<Operation *>>
+                        &loopToSharedMemref,
+                    const sycl::ReqdWorkGroupSize &reqdWorkGroupSize) {
   if (reqdWorkGroupSize.empty())
     return std::nullopt;
 
-  unsigned reqdLocalMemory = 0;
+  unsigned reqdSharedMemory = 0;
   for (auto &entry : loopToSharedMemref) {
-    unsigned loopReqdLocalMemory = 0;
+    unsigned loopReqdSharedMemory = 0;
     for (Operation *memref : entry.second) {
       sycl::AccessorType accTy =
           getAccessorType(cast<sycl::SYCLAccessorSubscriptOp>(memref));
-      loopReqdLocalMemory += getReqdLocalMemory(accTy, reqdWorkGroupSize);
+      loopReqdSharedMemory += getReqdSharedMemory(accTy, reqdWorkGroupSize);
     }
-    // Memref in one loop can reuse local memory allocated for another loop.
-    reqdLocalMemory = std::max(reqdLocalMemory, loopReqdLocalMemory);
+    // Memrefs in this loop can reuse shared memory allocated for another
+    // loop.
+    reqdSharedMemory = std::max(reqdSharedMemory, loopReqdSharedMemory);
   }
-  return reqdLocalMemory;
+  return reqdSharedMemory;
 }
 
-/// Create a GlobalOp for workgroup local memory.
-memref::GlobalOp getWorkGroupLocalMemory(gpu::GPUModuleOp module,
-                                         unsigned size) {
+/// Create a GlobalOp for workgroup shared memory.
+memref::GlobalOp getWorkGroupSharedMemory(gpu::GPUModuleOp module,
+                                          unsigned size) {
   assert(size != 0 && "Expecting non-zero size");
-  std::string name("WGLocalMem");
```

Comment:
If possible, I would like to use the same global variable name as LLVM.

Reply:
Fair request.